stan/math/opencl/prim/qr_Q.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef STAN_MATH_OPENCL_PRIM_QR_Q_HPP
+#define STAN_MATH_OPENCL_PRIM_QR_Q_HPP
+#ifdef STAN_OPENCL
+#include <stan/math/opencl/qr_decomposition.hpp>
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns the orthogonal factor of the fat QR decomposition
+ *
+ * @tparam T_m type of the matrix
+ * @param m Matrix.
+ * @return Orthogonal matrix with maximal columns
+ */
+template <typename T_m,
+          require_all_kernel_expressions_and_none_scalar_t<T_m>* = nullptr>
+inline matrix_cl<double> qr_Q(T_m&& m) {
+  check_nonzero_size("qr_Q(OpenCL)", "m", m);
+
+  matrix_cl<double> mat_eval = std::forward<T_m>(m);
+  matrix_cl<double> Q, R;
+
+  qr_decomposition_cl(m, Q, R);
+
+  auto Q_left
```

Comment:
What is the purpose of these two lines involving Q_left?

Reply:
`Q_left` is the part of Q that is affected by the signs of diagonal elements of R. So this is the part that needs to be changed when ensuring R has positive diagonal elements. So the first line defines this part and the second one selects either elements from this as they are or negated ones, depending on the diagonal of R.

Reply:
OK, so the two lines involving `Q_left` have the side effect of changing `Q`? Otherwise, I don't see why `Q` is returned rather than `Q_left`?

Reply:
Yes, Q_left is a view (it references Q's data), so modifying Q_left modifies Q.

Reply:
OK. Maybe that is self-evident to people who know the OpenCL interface.